### DISCUSSION TITLE
How to get remoteAddress from docker ?

### DISCUSSION THREAD

**Discussion by seepine** [fastify/fastify#5633] [Q&A] [answered]
I deploy my web app in docker
  - docker container: fastify-app, ip: 10.0.0.130

My ubuntu server ip: 172.18.51.2

My client request are
```
client -> nginx (ubuntuServer) proxy -> fastify-app
```
And fastify log 
```
{"remoteAddress":"10.0.0.2","remotePort":56022}
```

**Comment by seepine** [comment]
My nginx config
```conf
location ^~ / {
    proxy_pass http://127.0.0.1:8080; 
    proxy_set_header Host $host; 
    proxy_set_header X-Real-IP $remote_addr; 
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
    proxy_set_header REMOTE-HOST $remote_addr; 
    proxy_set_header Upgrade $http_upgrade; 
    proxy_set_header Connection "upgrade"; 
    proxy_set_header X-Forwarded-Proto $scheme; 
    proxy_http_version 1.1; 
    add_header X-Cache $upstream_cache_status; 
}
```

**Reply by metcoder95** [reply, marked as answer]
You need to set `trustProxy` to true while instantiating your Fastify app, so you can get the remote address when behind a reverse proxy.

[Ref](https://fastify.dev/docs/latest/Reference/Server/#trustproxy).

**Reply by seepine** [reply]
Hi, you mean code
```
Fastify({ trustProxy: true })
```
And it will got address from `x-forwarded-host` or `x-forwarded-proto` request headers?

**Reply by metcoder95** [reply]
Yes

**Reply by gurgunday** [reply]
Just be careful that `trustProxy: true` trusts **all** proxies, which means it can easily be spoofed if the proxies themselves are not configured properly, the ideal is to use a number that represents the number of proxies you have

**Reply by seepine** [reply]
> 请注意信任**所有**代理，这意味着如果代理本身配置不正确，它很容易被欺骗，理想的做法是使用一个代表您拥有的代理数量的数字`trustProxy: true`

I don't understand, I just want the logs to print out the client's IP address, the `trustProxy` has security issues?

**Reply by jsumners** [reply]
Please review https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For. We cannot educated every user on the specs.
